### PR TITLE
BOM-1045

### PR DESCRIPTION
**Pull request by Ayub-Khan**
-Added django 2.0,2.1,2.2 tests to travis
-Moved from djangorestframework-jwt to drf-jwt
-Updated Code files
-Updated Requirement files
-Updated Settings

### CODE REVIEW

**Comment by Ayub-Khan** [comment]
edx-platform test : https://github.com/edx/edx-platform/pull/23248

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,18 @@ env:
   - TOXENV=django111-drf38
   - TOXENV=django111-drf39
   - TOXENV=django111-drflatest
+  - TOXENV=django20-drf37
+  - TOXENV=django20-drf38
```
Comment (jmbowman) [inline .travis.yml]: Do we still need the DRF 3.7 and 3.8 tests?  Nearly all of our services have already upgraded to at least 3.9.4, and the others should before upgrading this.

File: edx_rest_framework_extensions/auth/jwt/authentication.py [hunk 2]
```diff
@@ -85,6 +83,8 @@ def authenticate(self, request):
             # CSRF passed validation with authenticated user
             return user_and_auth
 
+        except jwt.InvalidTokenError:
+            raise exceptions.AuthenticationFailed()
```
Comment (jmbowman) [inline edx_rest_framework_extensions/auth/jwt/authentication.py]: Thanks, that should be a pretty clean solution for this.

File: requirements/base.in [hunk 3]
```diff
@@ -2,11 +2,11 @@
 
 Django>=1.11
 djangorestframework>=3.2.0,<3.10
-djangorestframework-jwt>=1.7.2,<2.0.0
+drf-jwt<1.15.0
```
Comment (jmbowman) [inline requirements/base.in]: Please move the version constraint to `constraints.txt` with an explanatory comment, and leave just `drf-jwt` here.

File: edx_rest_framework_extensions/auth/jwt/tests/test_middleware.py [hunk 4]
```diff
@@ -276,15 +276,15 @@ def setUp(self):
         ('/loginredirectifunauthenticated/', True, 200),
         ('/isauthenticatedandloginredirect/', False, 302),
         ('/isauthenticatedandloginredirect/', True, 200),
-        ('/isauthenticated/', False, 403),
-        ('/isauthenticated/', True, 403),
+        ('/isauthenticated/', False, 401),
+        ('/isauthenticated/', True, 401),
```
Comment (jmbowman) [inline edx_rest_framework_extensions/auth/jwt/tests/test_middleware.py]: Do you know which specific upgrade triggered this?  Just wondering if it'll be a problem for the mobile apps, or if we already covered that case with the oauthlib upgrade testing.

File: requirements/base.in (hunk 3 above)
Comment (Ayub-Khan) [reply requirements/base.in]: 1.15.0 causes django22 test failures. will try to fix them tomorrow.

File: edx_rest_framework_extensions/auth/jwt/tests/test_middleware.py (hunk 4 above)
Comment (Ayub-Khan) [reply edx_rest_framework_extensions/auth/jwt/tests/test_middleware.py]: Im not sure. I think this might be because of the way tests are configured.

**Pull request merged**